core/server/master/src/main/java/alluxio/master/file/UfsSyncChecker.java

```diff
@@ -86,8 +87,10 @@ public void checkDirectory(InodeDirectory inode, AlluxioURI alluxioUri)
         .filter(ufsStatus -> !PathUtils.isTemporaryFileName(ufsStatus.getName()))
         .toArray(UfsStatus[]::new);
     Arrays.sort(ufsChildren, Comparator.comparing(UfsStatus::getName));
+    Inode[] alluxioChildren = Iterables.toArray(mInodeStore.getChildren(inode), Inode.class);
+    Arrays.sort(alluxioChildren);
     int ufsPos = 0;
-    for (Inode alluxioInode : mInodeStore.getChildren(inode)) {
+    for (Inode alluxioInode : alluxioChildren) {
```

Comment:
Maybe add `getSortedChildren` API if we anticipate this pattern will be used in other places as well?

Reply:
I think it's uncommon to need them sorted. Even here, we could use hashing as an alternative to sorting